### ISSUE TITLE
does monehTweets work with current versionof Twitter

### ISSUE THREAD

**Issue by yreko** [coldfumonkeh/monkehTweets#27]
Hello, hope you're couped after marathon!

I am not sure if its works with Twitter anymore. I tried all, I either get -- 

Element AUTHURL is undefined in AUTHSTRUCT   when I load up index.cfm, 

Or when try to d get direct authorization I get -- JSON parsing failure at character 1:'C' in Connection Failure 
Any ideas by a chance?
there is so little info avail for ColdFusion and Twitter, that I am not sure whats wrong.

**Comment by Mark-Hetherington** [comment]
The "Connection Failure" issue is generally due to certificate issues on your Cold Fusion server. Have you looked at issue #16 ?

**Comment by yreko** [comment]
Thank you for quick response!
Yes i did read that thread.  
You havnt mentioned any SSL certificates in monehTweets, so i am a bit of confused.
However, I am not sure where to obtain twitter certificate, most likely I am missing something really simple. 
I assume I need to find twitter.cer?
Where I can find it? because https://api.twitter.com is not working and there is no mention of it on app pages at https://apps.twitter.com/ 
Any help?
:-)

**Comment by Mark-Hetherington** [comment]
The twitter certificate is given to a connecting machine as part of the SSL connection. My understanding is by default the Cold Fusion server will try to validate an SSL certificate when connecting to an SSL server. If your Cold Fusion server doesn't validate it, it will often close the connection. You might need to install the CA certificates that signed the twitter certificate for it to validate. See also issue #13 and #20.

To confirm an SSL connection you could try testing a cfhttp request to the API to see if it will return an error also. If it does then you will need to resolve that first - it is a platform issue outside the scope of the monkehtweets library.

In terms of the AUTHURL is undefined error, it looks like that has also been logged previously as issue #7 .

**Comment by yreko** [comment]
i tried it yesterday - when try this 
<cfhttp url="https://twitter.com/">
</cfhttp> --- I wont get any errors
so isnt that mean that I dont have certificate issue?

as for callback URL  smth like http://mysite.com/m/authorize.cfm  should be sufficient, right?
i read all those threads before I contacted you.

**Comment by yreko** [comment]
i tried it yesterday - when try this
`<cfhttp url="https://twitter.com/">
</cfhttp>`
--- I wont get any errors
so isnt that mean that I dont have certificate issue?

as for callback URL smth like http://mysite.com/m/authorize.cfm should be sufficient, right?
i read all those threads before I contacted you.

**Comment by Mark-Hetherington** [comment]
The twitter site may use a different certificate to the API. If a CFHTTP request to e.g. "https://api.twitter.com/1.1/search/tweets.json" works without a connection failure, then that should exclude an SSL issue. To further troublesshoot I'd be inclined to add some logic to the part of the monkehtweets library that makes the request and CFDUMP out what the result is from the cfhttp request to get a bit more detail on the failure, or look at how to replicate the failure as a minimal test case outside of the library.

We are currently using the Monkehtweets library to retrieve a users timeline, so it is definitely working on the current twitter API.

**Comment by yreko** [comment]
1. `<cfhttp url="https://api.twitter.com/1.1/search/tweets.json">
</cfhttp>`  works without errors
yes it does not seem certificate issue, I getting puzzled! 

i get connection failure if i read `cfhttp.fileContent`
from index.cfm from monkehTweets   the value for authStruct.succees is false
but if go to cfc(s) all arguments are passed

**Comment by coldfumonkeh** [comment]
Hi @yreko and @Mark-Hetherington 

I'll take a look at this today.

**Comment by coldfumonkeh** [comment]
@yreko what version of ColdFusion are you running? What JDK version?

There was a conversation about the SSL / crt issue on a forum. The following post (linked) was from a fellow developer who resolved the cert issue on CF9. This may help you.

https://twittercommunity.com/t/discussion-for-rest-api-ssl-certificate-update/12226/31

**Comment by yreko** [comment]
On the main machine I have 9.0.1 JDK 1.7 
Gonna put fix 4 and follow the cert upgrade suggested by coldfusioner (above link), after I back everything up in the upcoming hour.
I'll let you know the result, because I am getting puzzled with this really not that complicated issue with connection failure to Twitter. 
Thank you so much for response.

**Comment by yreko** [comment]
I had fix 4 installed 2 years ago just forgot about it.

**Comment by yreko** [comment]
i made it worked, and as usual the solution was so-oo-o basic!
That was a certificate issue, and the link helped me. `https://twittercommunity.com/t/discussion-for-rest-api-ssl-certificate-update/12226/31`
in my case the two certificates VeriSign Class 2 Public Primary CA - G2 --- VeriSign Class 3 Public Primary CA - G3 ----  I installed a few years ago, and forgot about them 
That's why I never got errors by doing simple cfhttp call!
And that's why I was splitting hair because I felt like an idiot! LOL
--> After I installed this morning VeriSign Class 3 Primary CA - G5  --- my server is able to connect to Twitter and everything works!
Awesome!
Thank a lot guys for rapid and great response!
